src/main/java/xyz/ryozuki/helperbot/CommandHandler.java

```diff
@@ -6,62 +6,68 @@
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
 
-import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 public class CommandHandler implements CommandExecutor {
+
+    private final List<String> commands;
     private HelperBot plugin;
-    static String[] commands = new String[] {"setname", "reload"};
 
-    CommandHandler(HelperBot plugin) {
-        this.plugin = plugin;
+    CommandHandler() {
+        this.plugin = HelperBot.getInstance();
+
+        commands = new ArrayList<>();
+        commands.add("setname");
+        commands.add("reload");
+    }
+
+    public List<String> getCommands() {
+        return Collections.unmodifiableList(commands);
     }
 
     @Override
     public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
-        if(args.length < 1) {
+        if (args.length < 1) {
             sender.sendMessage("§aSubcommands: " + String.join(", ", commands));
             return true;
         }
-        String subcmd = args[0];
+
+        String subCmd = args[0];
         args = (String[]) ArrayUtils.remove(args, 0);
 
         String noPermissionText = "§4You don't have permission to execute this command.§r";
 
-        switch (subcmd) {
-            case "setname": {
-                if(!sender.hasPermission("helperbot.setname")) {
-                    sender.sendMessage(noPermissionText);
-                    break;
-                }
-                if(args.length < 1) {
-                    sender.sendMessage("§cMissing argument <name>");
-                    break;
-                }
-                plugin.setBotName(String.join(" ", args));
-                sender.sendMessage(
-                        String.format(
-                                "§aBot name succesfully set to:§r '%s§r'",
-                                ChatColor.translateAlternateColorCodes('&',
-                                        plugin.getBotName())
-                        )
-                );
-                break;
+        if (subCmd.equalsIgnoreCase("setname")) {
```

Comment:
Keep the switch case, it avoids repetitive code, just do `switch(subCmd.toLowerCase()) {`